### ISSUE TITLE
A `torch+julia 1.6` image

### ISSUE THREAD

**Issue by williamFalcon** [gridai/gridai#26]
Can we add in a new image type which adds Julia to one of your existing docker files?
Basically, take something like https://github.com/docker-library/julia/blob/master/1.6/alpine3.13/Dockerfile  or https://github.com/docker-library/julia/blob/master/1.6/buster/Dockerfile , change the from to your own, and then copy that text over to a new image?  There isn't that much to do in it:
ENV JULIA_PATH /usr/local/julia
ENV PATH $JULIA_PATH/bin:$PATH
# https://julialang.org/juliareleases.asc
# Julia (Binary signing key) <buildbot@julialang.org>
ENV JULIA_GPG 3673DF529D9049477F76B37566E3C7DC03D6E495
# https://julialang.org/downloads/
ENV JULIA_VERSION 1.6.0
RUN set -eux; \
	\
	apk add --no-cache --virtual .fetch-deps gnupg; \
	\
# https://julialang.org/downloads/#julia-command-line-version
# https://julialang-s3.julialang.org/bin/checksums/julia-1.6.0.sha256
# this "case" statement is generated via "update.sh"
	apkArch="$(apk --print-arch)"; \
	case "$apkArch" in \
# amd64
		x86_64) tarArch='x86_64'; dirArch='x64'; sha256='297e04c05e70da8fa575a2e2dd8151d57324b17870f6b60624e9e5629af8c939' ;; \
		*) echo >&2 "error: current architecture ($apkArch) does not have a corresponding Julia binary release"; exit 1 ;; \
	esac; \
	\
	folder="$(echo "$JULIA_VERSION" | cut -d. -f1-2)"; \
	wget -O julia.tar.gz.asc "https://julialang-s3.julialang.org/bin/musl/${dirArch}/${folder}/julia-${JULIA_VERSION}-musl-${tarArch}.tar.gz.asc"; \
	wget -O julia.tar.gz     "https://julialang-s3.julialang.org/bin/musl/${dirArch}/${folder}/julia-${JULIA_VERSION}-musl-${tarArch}.tar.gz"; \
	\
	echo "${sha256} *julia.tar.gz" | sha256sum -c -; \
	\
	export GNUPGHOME="$(mktemp -d)"; \
	gpg --batch --keyserver ha.pool.sks-keyservers.net --recv-keys "$JULIA_GPG"; \
	gpg --batch --verify julia.tar.gz.asc julia.tar.gz; \
	command -v gpgconf > /dev/null && gpgconf --kill all; \
	rm -rf "$GNUPGHOME" julia.tar.gz.asc; \
	\
	mkdir "$JULIA_PATH"; \
	tar -xzf julia.tar.gz -C "$JULIA_PATH" --strip-components 1; \
	rm julia.tar.gz; \
	\
	apk del --no-network .fetch-deps; \
	\
# smoke test
	julia --version
I would base it off of the torch or lightning ones since those should have the GPU drivers setup correctly/etc.

**Comment by williamFalcon** [comment]
A small addition.  If it is relatively easy for you to to add in a step in the image creation process to instantiate and precompile the packages (which can sometimes take a long time on julia since it is a compiled language) then all you need to do is is (1) check if a Project.toml exists in the root of the repo and if it does, execute the following:
julia --project=@. -e 'using Pkg; Pkg.instantiate()'
See https://github.com/jlperla/julia_binder_test  for a test repo to try it out with.
Finally, if you wanted to get fancy and support calling julia scripts then the rule would be that if any file has a .jl extension to call it you would go julia --project=@. demo.jl --num 5
so basically you just replace what would have been the python XXX with julia --project=@. XXX

**Comment by sunitaprakash** [comment]
Support for Julia Language has been added in the recent release
